sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala

```diff
@@ -253,7 +253,7 @@ private[sql] object InferSchema {
    * Returns the most general data type for two given data types.
    */
   def compatibleType(t1: DataType, t2: DataType): DataType = {
-    TypeCoercion.findTightestCommonTypeOfTwo(t1, t2).getOrElse {
+    TypeCoercion.findTightestCommonType(t1, t2).getOrElse {
```

Comment:
In CSVInferSchema.scala, it sounds like we are also calling the same function. Could you check whether we can directly call it, instead of duplicating the code? (I did not read it carefully)

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala#L172-L217